models/WebsiteSetting.php

```diff
@@ -274,6 +278,27 @@ public function delete(): void
             unset(self::$nameIdMappingCache[$nameCacheKey]);
         }
 
+        $event = new WebsiteSettingEvent($this);
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::PRE_DELETE);
+
         $this->getDao()->delete();
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::POST_DELETE);
+    }
+
+    public function save(): void
+    {
+        $event = new WebsiteSettingEvent($this);
+
+        $this->dispatchEvent($event, $this->id === null ?
+            WebsiteSettingEvents::PRE_ADD : WebsiteSettingEvents::PRE_UPDATE
+        );
+
+        $this->getDao()->save();
+
+        $this->dispatchEvent($event, $this->id === null ?
+            WebsiteSettingEvents::POST_ADD : WebsiteSettingEvents::POST_UPDATE
```

Comment:
Would it be better to use the updated Website setting for the post events so that they include the setting ID or some changes done by the pre update events?

Reply:
The `WebsiteSetting` object is mutable, so everything should be available in the "post" events, right?

Reply:
Wait, that means you have to do the `$this->id === null` check *before* calling `save()` and assign it to a variable that can be checked afterwards, because after that it should *always* have an ID and `POST_ADD` would never be called 🤔

Reply:
Oh, yes, you’re right—the event was never dispatched. I’ve fixed it.